src/main/groovy/net/neoforged/gradleutils/GradleUtilsExtension.groovy

```diff
@@ -99,4 +113,46 @@ abstract class GradleUtilsExtension {
     Action<? extends MavenArtifactRepository> getMaven() {
         return GradleUtils.getForgeMaven()
     }
+
+    void setupCentralPublishing() {
+        if (project.rootProject.name != project.name) {
+            throw new UnsupportedOperationException('The nexus publishing plugin can only be applied on the root project!')
```

Comment:
Is this a limitation of the nexus publishing plugin itself, or a limitation GU is imposing? 

What happens if a project using GradleUtils and calls this for signing is `includeBuild`ed into another project? Does this cause the whole build to fail because of the root project mismatch?

Reply:
It's a limitation of the nexus plugin, it must only be applied to the root project.

Reply:
Does this mean we can't `includeBuild` any NF project that publishes to Maven Central, for use in another project for rapid development, because the Nexus publishing plugin prevents that?

Reply:
It should not mean that, the root project of included builds doesn't change in a composite build scenario.

Reply:
Good, good.